analyzers/src/SonarAnalyzer.Common/Helpers/SonarLintXmlReader.cs

```diff
@@ -0,0 +1,114 @@
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO;
+using System.Text;
+using System.Xml;
+using System.Xml.Serialization;
+using Microsoft.CodeAnalysis.Text;
+
+namespace SonarAnalyzer.Helpers;
+
+public class SonarLintXmlReader
+{
+    public static readonly SonarLintXmlReader Empty = new(null, LanguageNames.CSharp);
+
+    private readonly SonarLintXml sonarLintXml;
+    private readonly string propertyLanguage;
+
+    private SonarLintXmlSettingsReader settings;
+    public SonarLintXmlSettingsReader Settings => settings ??= new(sonarLintXml, propertyLanguage);
+
+    public SonarLintXmlReader(SourceText originalXml, string language)
+    {
+        sonarLintXml = originalXml == null ? SonarLintXml.Empty : ParseContent(originalXml);
+        propertyLanguage = language;
+    }
+
+    private static SonarLintXml ParseContent(SourceText sonarProjectConfig)
+    {
+        try
+        {
+            var serializer = new XmlSerializer(typeof(SonarLintXml));
+            var byteArray = Encoding.UTF8.GetBytes(sonarProjectConfig.ToString());
+            var stream = new MemoryStream(byteArray);
+            using var sr = new StreamReader(stream, Encoding.UTF8, false);
+            using var reader = XmlReader.Create(sr);
+            return (SonarLintXml)serializer.Deserialize(reader);
+        }
+        catch
+        {
+            return SonarLintXml.Empty;
+        }
+    }
+}
+
+public class SonarLintXmlSettingsReader
+{
+    private readonly SonarLintXml sonarLintXml;
+    private readonly string propertyLanguage;
+
+    private bool? ignoreHeaderComments;
+    public bool? IgnoreHeaderComments => ignoreHeaderComments ??= ReadBoolean(ReadProperty($"sonar.{propertyLanguage}.ignoreHeaderComments"));
+
+    private bool? analyzeGeneratedCode;
+    public bool AnalyzeGeneratedCode => analyzeGeneratedCode ??= ReadBoolean(ReadProperty($"sonar.{propertyLanguage}.analyzeGeneratedCode"));
+
+    private bool? ignoreIssues;
+    public bool IgnoreIssues => ignoreIssues ??= ReadBoolean(ReadProperty($"sonar.{propertyLanguage}.roslyn.ignoreIssues"));
+
+    private string suffixes;
+    public string Suffixes => suffixes ??= ReadString(ReadProperty($"sonar.{propertyLanguage}.file.suffixes"));
```

Comment:
These two properties are not used in our codebase, so there is no need to add a way to read them. We don`t want to maintain code that is not used anywhere.